sdk/lib/drivers/sound/mmixer/mixer.c

```diff
@@ -836,18 +836,19 @@ MMixerInitialize(
             }
             else
             {
-                DPRINT1("Failed to enumerate device %lu\n", DeviceIndex);
+                DPRINT1("Failed to enumerate device %lu, Status %x\n", DeviceIndex, Status);
 
-                /* TODO cleanup */
-                return Status;
+                /* ignore error and continue */
             }
         }
         else
         {
             /* create a mixer data entry */
             Status = MMixerCreateMixerData(MixerContext, MixerList, DeviceIndex, DeviceName, hMixer, hKey);
             if (Status != MM_STATUS_SUCCESS)
-                break;
+                DPRINT1("Failed to allocate mixer data for device %lu, Status %x\n", DeviceIndex, Status);
```

Comment:
Either explicitly check for `MM_STATUS_NO_MEMORY` or don't make assumptions:
```suggestion
                DPRINT1("MMixerCreateMixerData() failed for device %lu, Status %x\n", DeviceIndex, Status);
```

Reply:
Applied, thanks.